spec/support/shared_entity_specs.rb

```diff
@@ -70,3 +70,19 @@ def validate_values(value, parsed_value, type)
     end
   end
 end
+
+shared_examples "a relayable Entity" do
+  let(:instance) { described_class.new(data) }
+
+  context "signatures generation" do
+    it "computes correct signatures for the entity" do
+      xml = DiasporaFederation::Salmon::XmlPayload.pack(instance)
+      hash = DiasporaFederation::Salmon::XmlPayload.hash(xml.at_xpath("post/*[1]"))
```

Comment:
ah, see it, but does this even work? `instance.author_signature` will not be updated with `pack`. So I think you can use `to_h` instead of `hash`, but you need to parse the signatures from the xml. And you sould use a `data` without updated signature as initial data.